pkg/dashboard/keyvisual/input/core.go

```diff
@@ -0,0 +1,112 @@
+// Copyright 2020 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package input
+
+import (
+	regionPKG "github.com/pingcap-incubator/tidb-dashboard/pkg/keyvisual/region"
+	"github.com/pingcap/pd/server"
+	"github.com/pingcap/pd/server/cluster"
+	"github.com/pingcap/pd/server/core"
+)
+
+const limit = 1024
+
+// RegionsInfo implements the interface regionPKG.RegionsInfo for [] * core.RegionInfo.
+type RegionsInfo []*core.RegionInfo
+
+// Len returns the number of regions.
+func (rs RegionsInfo) Len() int {
+	return len(rs)
+}
+
+// GetKeys returns the sorted endpoint keys of all regions.
+// Fixme: StartKey may not be equal to the EndKey of the previous region
```

Comment:
I think we should fix it ASAP?

Reply:
This problem occurs because clusterScan is executed sequentially. The problem of holes is small and has little effect. And it cannot be completely solved. If you choose to insert an empty region in the hole, the keys and values cannot be aligned with the regions, I think it is worth the loss.